homeassistant/components/acmeda/cover.py

```diff
@@ -0,0 +1,230 @@
+"""Support for Acmeda Roller Blinds."""
+import asyncio
+import logging
+from time import monotonic
+
+import aiopulse
+import async_timeout
+
+from homeassistant.components.cover import ATTR_POSITION, CoverDevice
+
+from .base import AcmedaBase
+from .const import DOMAIN
+from .helpers import remove_devices
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up the Acmeda Rollers from a config entry."""
+    hub = hass.data[DOMAIN][config_entry.data["host"]]
+    cur_covers = {}
+    # cur_groups = {}
+
+    # Acmeda updates all covers via a single API call.
+    #
+    # If we call a service to update 2 covers, we only want the API to be
+    # called once.
+    #
+    # The throttle decorator will return right away if a call is currently
+    # in progress. This means that if we are updating 2 covers, the first one
+    # is in the update method, the second one will skip it and assume the
+    # update went through and updates it's data, not good!
+    #
+    # The current mechanism will make sure that all covers will wait till
+    # the update call is done before writing their data to the state machine.
+    #
+    # An alternative approach would be to disable automatic polling by Home
+    # Assistant and take control ourselves. This works great for polling as now
+    # we trigger from 1 time update an update to all entities. However it gets
+    # tricky from inside async_turn_on and async_turn_off.
+    #
+    # If automatic polling is enabled, Home Assistant will call the entity
+    # update method after it is done calling all the services. This means that
+    # when we update, we know all commands have been processed. If we trigger
+    # the update from inside async_turn_on, the update will not capture the
+    # changes to the second entity until the next polling update because the
+    # throttle decorator will prevent the call.
+
+    progress = None
+    cover_progress = set()
+    group_progress = set()
+
+    async def request_update(is_group, object_id):
+        """Request an update.
+
+        We will only make 1 request to the server for updating at a time. If a
```

Comment:
Can this be moved into the `aiopulse` package so Home Assistant doesn't have to know about this complexity?

Reply:
Good suggestion, have enhanced aiopulse so it will notify the integration of any changes to the hub configuration. Now using `helpers/dispatcher.py` to notify the entities that they need to check for new instances.